jscomp/runtime/js_array.ml

```diff
@@ -25,216 +25,193 @@
+external sortInPlace : 'a t -> 'a t = "sort" [@@mel.send]
 
-external sortInPlaceWith : (('a -> 'a -> int)[@mel.uncurry]) -> 'this = "sort"
-[@@mel.send.pipe: 'a t as 'this]
+external sortInPlaceWith : 'a t -> f:(('a -> 'a -> int)[@mel.uncurry]) -> 'a t
+  = "sort"
+[@@mel.send]
 
-external spliceInPlace : pos:int -> remove:int -> add:'a array -> 'this
+external spliceInPlace : 'a t -> pos:int -> remove:int -> add:'a array -> 'a t
   = "splice"
-[@@mel.send.pipe: 'a t as 'this] [@@mel.splice]
+[@@mel.send] [@@mel.splice]
 
-external removeFromInPlace : pos:int -> 'this = "splice"
-[@@mel.send.pipe: 'a t as 'this]
+external removeFromInPlace : 'a t -> pos:int -> 'a t = "splice" [@@mel.send]
 
-external removeCountInPlace : pos:int -> count:int -> 'this = "splice"
-[@@mel.send.pipe: 'a t as 'this]
+external removeCountInPlace : 'a t -> pos:int -> count:int -> 'a t = "splice"
+[@@mel.send]
 (* screwy naming, but screwy function *)
 
-external unshift : 'a -> int = "unshift" [@@mel.send.pipe: 'a t as 'this]
-
-external unshiftMany : 'a array -> int = "unshift"
-[@@mel.send.pipe: 'a t as 'this] [@@mel.splice]
-
-(* Accessor functions
-*)
-external append : 'a -> 'this = "concat"
-[@@mel.send.pipe: 'a t as 'this]
-[@@deprecated "append is not type-safe. Use `concat` instead, and see #1884"]
-
-external concat : 'this -> 'this = "concat" [@@mel.send.pipe: 'a t as 'this]
-
-external concatMany : 'this array -> 'this = "concat"
-[@@mel.send.pipe: 'a t as 'this] [@@mel.splice]
+external unshift : 'a t -> value:'a -> int = "unshift" [@@mel.send]
 
-(* TODO: Not available in Node V4  *)
-external includes : 'a -> bool = "includes"
-[@@mel.send.pipe: 'a t as 'this]
-(** ES2016 *)
+external unshiftMany : 'a t -> values:'a array -> int = "unshift"
+[@@mel.send] [@@mel.splice]
 
-external indexOf : 'a -> int = "indexOf" [@@mel.send.pipe: 'a t as 'this]
+(** Accessor functions *)
 
-external indexOfFrom : 'a -> from:int -> int = "indexOf"
-[@@mel.send.pipe: 'a t as 'this]
+external concat : 'a t -> other:'a t -> 'a t = "concat" [@@mel.send]
 
-external join : 'a t -> string = "join"
-[@@mel.send] [@@deprecated "please use joinWith instead"]
+external concatMany : 'a t -> arrays:'a t array -> 'a t = "concat"
+[@@mel.send] [@@mel.splice]
 
-external joinWith : string -> string = "join" [@@mel.send.pipe: 'a t as 'this]
+external includes : 'a t -> value:'a -> bool = "includes"
+[@@mel.send]
+(** ES2015 *)
 
-external lastIndexOf : 'a -> int = "lastIndexOf"
-[@@mel.send.pipe: 'a t as 'this]
+external indexOf : 'a t -> value:'a -> int = "indexOf" [@@mel.send]
 
-external lastIndexOfFrom : 'a -> from:int -> int = "lastIndexOf"
-[@@mel.send.pipe: 'a t as 'this]
+external indexOfFrom : 'a t -> value:'a -> from:int -> int = "indexOf"
+[@@mel.send]
 
-external lastIndexOf_start : 'a -> int = "lastIndexOf"
-[@@mel.send.pipe: 'a t as 'this] [@@deprecated "Please use `lastIndexOf"]
+external join : 'a t -> sep:string -> string = "join" [@@mel.send]
```

Comment:
Is it possible to have `sep` as optional? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join#separator